presto-postgresql/src/test/java/io/prestosql/plugin/postgresql/TestPostgreSqlTypeMapping.java

```diff
@@ -543,19 +568,50 @@ private void addTimestampTestIfSupported(DataTypeTest tests, boolean legacyTimes
         };
     }
 
-    private static void checkIsGap(ZoneId zone, LocalDateTime dateTime)
+    @Test(dataProvider = "testTimestampWithTimeZoneDataProvider")
+    public void testTimestampWithTimeZone(boolean insertWithPresto)
     {
-        verify(isGap(zone, dateTime), "Expected %s to be a gap in %s", dateTime, zone);
-    }
+        DataType<ZonedDateTime> dataType;
+        DataSetup dataSetup;
+        if (insertWithPresto) {
+            dataType = prestoTimestampWithTimeZoneDataType();
+            dataSetup = prestoCreateAsSelect("test_timestamp_with_time_zone");
+        }
+        else {
+            dataType = postgreSqlTimestampWithTimeZoneDataType();
+            dataSetup = postgresCreateAndInsert("tpch.test_timestamp_with_time_zone");
+        }
 
-    private static boolean isGap(ZoneId zone, LocalDateTime dateTime)
```

Comment:
Doing this refactor first as a separate commit will make the diff cleaner

Reply:
Sorry for that. Do you want me to separate that, or you've already made your way through?